src/main/java/uk/gov/companieshouse/controller/DissolutionController.java

```diff
@@ -136,4 +137,33 @@ public DissolutionPatchResponse patchDissolutionApplication(
             throw new NotFoundException();
         }
     }
+
+    @PatchMapping("/directors/{director-id}")
+    @ResponseStatus(HttpStatus.OK)
+    public DissolutionPatchResponse patchDissolutionDirector(
+            @RequestHeader("ERIC-Authorised-User") String authorisedUser,
+            @PathVariable("company-number") final String companyNumber,
+            @PathVariable("director-id") final String directorId,
+            @Valid @RequestBody final DissolutionPatchDirectorRequest body
+    ) {
+
+        if (!dissolutionService.doesDissolutionRequestExistForCompanyByCompanyNumber(companyNumber)
+                || !dissolutionService.doesDirectorExist(companyNumber, directorId)) {
+            throw new NotFoundException();
+        }
+
+        if (!dissolutionService.isDirectorPendingApproval(companyNumber, directorId)) {
+            throw new BadRequestException("Director is not pending approval");
+        }
+
+        if (!dissolutionService.doesEmailBelongToApplicant(companyNumber, getEmail(authorisedUser))) {
```

Comment:
Same here, i can imagine this can be combined to reduce db calls.

Reply:
Do you want this function to operate on a already fetched dissolution, or how would you see this?

Reply:
```
final Optional<String> error = dissolutionService.checkPatchDirectorConstraints(companyNumber, directorId, getEmail(authorisedUser));
if (error.isPresent()) {
  throw new BadRequestException(error.get());
}
```

then in the service:
```
final Dissolution dissolution = repository.findByCompanyNumber(companyNumber).orElseThrow(...);

if (!isDirectorPendingApproval(dissolution, directorId)) {
  return Optional.of("Director is not pending approval");
}

if (!doesEmailBelongToApplicant(dissolution, email)) {
  return Optional.of("Only the applicant can update signatory");
}

return Optional.empty();
```
I'd make the errors variables too btw